flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableEnvironmentITCase.scala

```diff
@@ -808,6 +808,162 @@ class TableEnvironmentITCase(tableEnvName: String, isStreaming: Boolean) extends
     listener.close()
   }
 
+  // `SHOW COLUMNS ......` syntax of flink SQL test cases.
+  @Test
+  def testShowColumnsSyntaxOfSQL(): Unit = {
+
+    initTableAndView()
+    // Tests `SHOW COLUMNS FROM TABLE`.
+    showAllColumnsFromTable()
+    showColumnsWithLikeClauseFromTable()
+    showColumnsWithNotLikeClauseFromTable()
+
+    // Tests `SHOW COLUMNS FROM VIEW`.
+    showAllColumnsFromView()
+    showColumnsWithLikeClauseFromView()
+    showColumnsWithNotLikeClauseFromView()
+
+  }
+
+  private def initTableAndView(): Unit = {
+    val createClause: String =
+      s"""
+         |CREATE TABLE IF NOT EXISTS orders (
+         | `user` BIGINT NOT NULl,
+         | `product` VARCHAR(32),
+         | `amount` INT,
+         | PRIMARY KEY(`user`) NOT ENFORCED
+         |) """.stripMargin
+    var createWithClause: String =
+      s"""
+         |with (
+         | 'connector' = 'datagen'
+         |)""".stripMargin
+    if (!isStreaming) {
+      val sinkPath = _tempFolder.newFolder().toString
+      createWithClause =
+        s"""
+           |with (
+           |  'connector' = 'filesystem',
+           |  'path' = '$sinkPath',
+           |  'format' = 'testcsv'
+           |)""".stripMargin
+    }
+    tEnv.executeSql(createClause + createWithClause)
+    tEnv.executeSql("create view orders_view as select * from orders")
+  }
+
+  private def showAllColumnsFromTable(): Unit = {
+    val expectedResultStr: String = "[" +
+      "+I[user, BIGINT, false, PRI(user), null, null], " +
+      "+I[product, VARCHAR(32), true, null, null, null], " +
+      "+I[amount, INT, true, null, null, null]" +
+      "]"
```

Comment:
The expected result would be better to be Row list, we avoid to compare Rows by string.